src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -202,6 +202,9 @@ public ImmutableArray<NamedTypeSymbol> Interfaces
         /// list. This is not quite the same as "all interfaces of which this type is a proper
         /// subtype" because it does not take into account variance: AllInterfaces for
         /// IEnumerable&lt;string&gt; will not include IEnumerable&lt;object&gt;
+        ///
+        /// Note: When interfaces specified on the same inheritance level differ by tuple names only,
+        /// only the last one will be listed here.
```

Comment:
Btw, I confirmed this first vs. last problem was not visible previously. It would not show with object vs dynamic scenarios, since dynamic is apparently disallowed in interfaces.